lib/spack/spack/cmd/__init__.py

```diff
@@ -143,15 +150,34 @@ def parse_specs(args, **kwargs):
     except spack.spec.SpecParseError as e:
         msg = e.message + "\n" + str(e.string) + "\n"
         msg += (e.pos + 2) * " " + "^"
-        raise SpackError(msg)
+        err_msg = msg
 
     except spack.spec.SpecError as e:
-
         msg = e.message
         if e.long_message:
             msg += e.long_message
-
-        raise SpackError(msg)
+        err_msg = msg
+
+    if err_msg:
+        try:
+            specs = []
+            for arg in args:
+                with open(arg, 'r') as fd:
+                    specs.append(spack.spec.Spec.from_yaml(fd))
+            return specs
+        except EnvironmentError:
+            # If we were unable to find, open, read file, etc probably just
+            # means the string wasn't a file path to begin with.  Any other
+            # exception at this point would indicate the spec *was* provided
+            # as a file path, and we should fail through so the user can see
+            # the underlying issue.
+            pass
```

Comment:
I think you should probably raise here -- doesn't this mean that something was neither a valid spec or a filename?  If you skip it at this point I think you're allowing things like:

```
spack install <@#$> valid_filename.yaml
```

Am I missing something?